clients/js/src/compat.ts

```diff
@@ -500,20 +499,39 @@ class EnvelopeError extends Error {}
  * Note: MetaMask does not support Web3 methods it doesn't know about, so we have to
  * fall back to manually querying the default gateway.
  */
-async function inferRuntimePublicKeySource(
+async function fetchRuntimePublicKey(
   upstream: UpstreamProvider,
-): Promise<Parameters<typeof fetchRuntimePublicKey>[0]> {
+): Promise<Uint8Array> {
   const isSigner = isEthersSigner(upstream);
+  let chainId: number;
   if (isSigner || isEthersProvider(upstream)) {
-    return {
-      chainId: isSigner
-        ? await upstream.getChainId()
-        : (await upstream.getNetwork()).chainId,
-    };
+    if (
+      'provider' in upstream &&
+      upstream['provider'] &&
+      'send' in upstream['provider']
+    ) {
+      // first opportunistically try `send` from the provider
+      try {
+        const source = upstream['provider'] as {
+          send: (method: string, params: any[]) => Promise<any>;
+        };
+        const { key } = await source.send(OASIS_CALL_DATA_PUBLIC_KEY, []);
+        if (key) {
+          return arrayify(key);
+        }
+      } catch (e) {
```

Comment:
```suggestion
      } catch {
```
to avoid the unused var